File: Medium/198-house-robber/try-3.py
/*
 * Problem #198: House Robber
 * Difficulty: Medium
 * Submission: Try 3
 * status: Accepted
 * Language: python3
 * Date: 09/01/2025, 23:03:45
 * Link: https://leetcode.com/problems/house-robber/
 */

class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) ==1:
            return nums[0]
        l1 =[]
        l1.extend([nums[0],max(nums[0],nums[1])])
        for i in range(2,len(nums)):
            l1.append(max(l1[i-1],l1[i-2]+nums[i]))
        return l1[-1]

File: Easy/26-remove-duplicates-from-sorted-array/try-3.cpp
/*
 * Problem #26: Remove Duplicates from Sorted Array
 * Difficulty: Easy
 * Submission: Try 3
 * status: Accepted
 * Language: cpp
 * Date: 02/10/2024, 13:12:59
 * Link: https://leetcode.com/problems/remove-duplicates-from-sorted-array/
 */

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int h=1,i;
        for(i=0;i<nums.size()-1;i++){
            if(nums[i]!=nums[i+1]){
                nums[h]=nums[i+1];
                h++;
            }
        }
    return h;
    }
};
